### ISSUE TITLE
Feature Request and Duplicates Found

### ISSUE THREAD

**Issue by beano38** [andrebrait/1g1r-romset-generator#9]
**First a couple of feature requests:**

1. DS games have additional attributes like NDSi Enhanced, WiFi Kiosk and Save Data, any chance to add some additional flags to remove/include these?
2. Can you have it output a "new" dat file with the filtered set as an option?  Rather than just move/copy a source ROM set?  Reason is I want to generate a new XML that is suitable for use in HyperSpin, RocketLauncher and etc.  I've already written a python script/program that takes a No Intro dat file and generates a database XML in the format expected.  It also filters out any ROMs I do not have and the final list is a "curated" database of ROMs I have that match No Intro naming.

I'm going to attempt to modify the code and do this myself, though I'm not as skilled a programmer as you.  Figured I'd ask...

**Duplicates found in the NDS set.**

command:  python generate.py -r USA,EUR -d "Nintendo - Nintendo DS (Decrypted) (Parent-Clone) (20200319-065304).dat" --no-all

Who Wants to Be a Millionaire (Europe) (En,Fr,De,Es,It,Nl)
Who Wants to Be a Millionaire (USA)

World Cup of Pool (USA)
World Cup of Pool (Europe) (En,Fr,De,Es,It)

Not sure if there are others, but these are just scrolling up from the bottom with the output.  Figure you may want to look at it.

By the way, great repo, I've attempted to write something that will do this but haven't had much luck, at least not as comprehensive as what you've done.

**Comment by andrebrait** [comment]
Nice to hear you're enjoying it :-)

Yes, that could be a nice feature addition. Is the XML format you want 100% compatible with DATs?

**Comment by andrebrait** [comment]
And regarding the duplicates issue, could you please attach the DAT you're using?

**Comment by andrebrait** [comment]
As for adding those options for NDS ROMs: although I myself use the tool for NDS, I decided not to include options specific for those. If I start doing that too much, I'll have like 100k options.

Using the exclusion options should be easy enough for most people.

I'm also going to post a lost of recommended settings in the future. It'll include exclusions like those and an explanation why they're on that list.

**Comment by beano38** [comment]
> Yes, that could be a nice feature addition. Is the XML format you want 100% compatible with DATs?

Yes, for me, having the output dat equivelent to a filtered version of the NoIntro dat would work great.  That allows me to use this for ClrMAMEpro and some of the tools I've made.  I leverage the SHA and CRC values in the tools, rather than name.  I did see you added some of the hashing for moving/renaming tho.  Haven't tried that part out yet.

> Using the exclusion options should be easy enough for most people.

I'll give it a shot
[Nintendo - Nintendo DS (Decrypted) (Parent-Clone) (20200319-123850).zip](https://github.com/andrebrait/1g1r-romset-generator/files/4359547/Nintendo.-.Nintendo.DS.Decrypted.Parent-Clone.20200319-123850.zip)

**Comment by andrebrait** [comment]
I'm planning a feature called Output Mode which will be able to generate
different folder/archive structures and also output files to a custom DAT
:-)

But don't close this issue. I'll do that when I add the feature.

As for the duplicates, I checked the DAT and it's missing the Parent/Clone
information for those games. I have reported that on the DAT-o-matic
forums. I'll let you know when they fix it.

André Brait

**Comment by EvilDragon** [comment]
Hands up for outputting DAT files including a folder structure :)
This could be used with RomVault, which is one of the best rom managers but still missing 1G1R :)
